src/plugins/GLesmos/glesmosCanvas.ts

```diff
@@ -112,27 +160,128 @@ void main() {
 export function initGLesmosCanvas() {
   //================= INIT ELEMENTS =======================
   let c: HTMLCanvasElement = document.createElement("canvas");
-  let gl: WebGLRenderingContext = c.getContext("webgl2", {
+  let gl: WebGL2RenderingContext = c.getContext("webgl2", {
     // Disable premultiplied alpha
     // Thanks to <https://stackoverflow.com/a/12290551/7481517>
     premultipliedAlpha: false,
-  }) as WebGLRenderingContext;
+  }) as WebGL2RenderingContext;
+
+  let prevFramebuffer: WebGLFramebuffer | null;
+  let prevTexture: WebGLTexture | null;
+
+  let currFramebuffer: WebGLFramebuffer | null;
+  let currTexture: WebGLTexture | null;
+
+  let currFramebuffer2: WebGLFramebuffer | null;
+  let currTexture2: WebGLTexture | null;
+
+  let vao: WebGLVertexArrayObject | null;
 
   //================= GRAPH BOUNDS ======================
   let cornerOfGraph = [-10, -6];
   let sizeOfGraph = [20, 12];
 
   //======================= RESIZING STUFF =======================
 
+  let currentWidth = 0;
+  let currentHeight = 0;
+
+  let forceUpdateTransforms = false;
+
   let updateTransforms = (transforms: ViewportTransforms) => {
+    //console.log("Updated GLesmos transforms.");
     const w = transforms.pixelCoordinates.right;
     const h = transforms.pixelCoordinates.bottom;
     const p2m = transforms.pixelsToMath;
-    c.width = w;
-    c.height = h;
-    gl.viewport(0, 0, c.width, c.height);
     cornerOfGraph = [p2m.tx, p2m.sy * h + p2m.ty];
     sizeOfGraph = [p2m.sx * w, -p2m.sy * h];
+
+    const [fw, fh] = [Math.floor(w), Math.floor(h)];
+
+    if (currentWidth == fw && currentHeight == fh && !forceUpdateTransforms)
+      return;
+    forceUpdateTransforms = false;
+    console.log(fw, fh);
+    currentWidth = fw;
+    currentHeight = fh;
+
+    c.width = fw;
+    c.height = fh;
+
+    currFramebuffer2 = gl.createFramebuffer();
+    currTexture2 = gl.createTexture();
+    gl.bindTexture(gl.TEXTURE_2D, currTexture2);
+    gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST);
+    gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.NEAREST);
+    gl.texImage2D(
+      gl.TEXTURE_2D,
+      0,
+      gl.RGBA,
+      Math.floor(w / speed),
+      Math.floor(h / speed),
+      0,
+      gl.RGBA,
+      gl.UNSIGNED_BYTE,
+      null
+    );
```

Comment:
This chunk of code (and maybe a bit before / after) is repeated a few times with just different width/height/buffer/texture. It should be split into a function, something like `bindTextureAndBuffer(gl, buffer, texture)` (or `bindTextureAndBuffer(gl, frameData)` if you merge the buffer and texture into a single type as suggested around line 175). Splitting to a function helps avoid duplication and clarifies what the function is doing (comments on each duplicated block would also work ig, but worse).

Reply:
Fixed!